src/bitmessagekivy/baseclass/popup.py

```diff
@@ -0,0 +1,231 @@
+
+from bitmessagekivy.baseclass.common import toast
+
+from addresses import decodeAddress
+from debug import logger
+
+
+class AddressChangingLoader(Popup):
+    """Run a Screen Loader when changing the Identity for kivy UI"""
+
+    def __init__(self, **kwargs):
+        super(AddressChangingLoader, self).__init__(**kwargs)
+        Clock.schedule_once(self.dismiss_popup, 0.5)
+
+    def dismiss_popup(self, dt):
+        """Dismiss popups"""
+        self.dismiss()
+
+
+class AddAddressPopup(BoxLayout):
+    """Popup for adding new address to addressbook"""
+
+    valid = False
+
+    def __init__(self, **kwargs):
+        super(AddAddressPopup, self).__init__(**kwargs)
+
+    def checkAddress_valid(self, instance):
+        """Checking address is valid or not"""
+        my_addresses = (
+            App.get_running_app().root.ids.content_drawer.ids.btn.values)
+        add_book = [addr[1] for addr in kivy_helper_search.search_sql(
+            folder="addressbook")]
+        entered_text = str(instance.text).strip()
+        if entered_text in add_book:
+            text = 'Address is already in the addressbook.'
+        elif entered_text in my_addresses:
+            text = 'You can not save your own address.'
+        elif entered_text:
+            text = self.addressChanged(entered_text)
+
+        if entered_text in my_addresses or entered_text in add_book:
+            self.ids.address.error = True
+            self.ids.address.helper_text = text
+        elif entered_text and self.valid:
+            self.ids.address.error = False
+        elif entered_text:
+            self.ids.address.error = True
+            self.ids.address.helper_text = text
+        else:
+            self.ids.address.error = False
+            self.ids.address.helper_text = 'This field is required'
+
+    def checkLabel_valid(self, instance):
+        """Checking address label is unique or not"""
+        entered_label = instance.text.strip()
+        addr_labels = [labels[0] for labels in kivy_helper_search.search_sql(
+            folder="addressbook")]
+        if entered_label in addr_labels:
+            self.ids.label.error = True
+            self.ids.label.helper_text = 'Label name already exists.'
+        elif entered_label:
+            self.ids.label.error = False
+        else:
+            self.ids.label.error = False
+            self.ids.label.helper_text = 'This field is required'
+
+    def _onSuccess(self, addressVersion, streamNumber, ripe):
+        pass
+
+    def addressChanged(self, addr):
+        """Address validation callback, performs validation and gives feedback"""
+        status, addressVersion, streamNumber, ripe = decodeAddress(
+            str(addr))
+        self.valid = status == 'success'
+
+        if self.valid:
+            text = "Address is valid."
+            self._onSuccess(addressVersion, streamNumber, ripe)
+            return text
+
+        validation_dict = {
```

Comment:
this data is static, so I would make it a class variable, or a module variable, or a constant

Reply:
or a DataClass?